boot/mbed/src/flash_map_backend.cpp

```diff
@@ -91,15 +91,20 @@ int flash_area_open(uint8_t id, const struct flash_area** fapp) {
     fap->fa_device_id = 0; // not relevant
 
     mbed::BlockDevice* bd = flash_map_bd[id];
-    fap->fa_size = (uint32_t) bd->size();
 
+    int result = 0;
     /* Only initialize if this isn't a nested call to open the flash area */
     if (open_count[id] == 1) {
         MCUBOOT_LOG_DBG("initializing flash area %d...", id);
-        return bd->init();
-    } else {
-        return 0;
+        result = bd->init();
+        if(result) {
```

Comment:
Not sure how lax the Mbed OS coding style is, but this could use a space to match other code: `if (`. Also please rebase on latest master.